integration/env/env.go

```diff
@@ -0,0 +1,72 @@
+package env
+
+import (
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+// Env should be created using Builder.
+type Env struct {
+	Azure   Azure
+	Common  Common
+	Cluster Cluster
+}
+
+type BuilderConfig struct {
+	Logger micrologger.Logger
+}
+
+type Builder struct {
+	logger micrologger.Logger
+}
+
+func NewBuilder(config BuilderConfig) (*Builder, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	b := &Bilder{
+		logger: config.Logger,
+	}
+
+	return b, nil
+}
+
+func (b *Builder) Build() (Env, error) {
+	var common Common
+	{
+		// ...
+	}
+
+	var cluster Cluster
+	{
+		// ...
+	}
+
+	var azure Azure
+	{
+		c := azureBuilderConfig{
+			Logger: b.logger,
+
+			CircleBuildNumber: common.Circle.BuildNumber,
```

Comment:
Here is an example on how to pass a value from common which wasn't really possible with `init()` functions.